x-pack/plugins/observability/server/services/slo/slo_repository.ts

```diff
@@ -19,13 +19,29 @@ import { sloSchema } from '../../types/schema';
 
 export interface Criteria {
   name?: string;
+  indicatorType?: string;
 }
 
 export interface Pagination {
   page: number;
   perPage: number;
 }
 
+export enum SortingDirection {
```

Comment:
I recently learned enums are not ideal: https://youtu.be/jjMbPt_H3RQ

An alternative would be using a pojo with 'as const'.

Reply:
_Agree_ they are not great **when the value is not specified**, but as he even says himself at [2:15](https://youtu.be/jjMbPt_H3RQ?t=135), this declaration is safe from a code point of view (no extra keys/values created). He just doesn't like the declaration, i.e. `Key = "Value"` but that's subjective at best. 
Moreover, his solution with POJO requires the exact same declaration, i.e. `Key = "Value"`, and some extra boilerplate for the type.

IMO using a `as const` is 1. as verbose as enum and 2. requires more typing boilerplate. 

That being said, I'm going to change the code to use `as const`, and see how the code evolves 🤷🏻

Reply:
In terms of verbosity I agree both approaches are the same. 

For me, it's the frankly [bizarre runtime behavior of enums](https://www.youtube.com/watch?v=0fTdCSH_QEU) and that one of the stewards of TS declared that the implementation of the enum feature in the language was not ideal and they would have done differently if they could redo it.

Reply:
I wonder if assertNever trick works with the as const method, I'll give it a try later.